patches/0004-Add-OpenSSL-crypto-backend.patch

```diff
@@ -550,6 +560,54 @@ index f8485dc3ca1c29..9c1d3d279c472f 100644
  
  package fipsonly
  
+diff --git a/src/crypto/tls/key_schedule.go b/src/crypto/tls/key_schedule.go
+index d7f082c9ee1e04..ee9d4650e3523b 100644
+--- a/src/crypto/tls/key_schedule.go
++++ b/src/crypto/tls/key_schedule.go
+@@ -7,9 +7,11 @@ package tls
+ import (
+ 	"crypto/ecdh"
+ 	"crypto/hmac"
++	boring "crypto/internal/backend"
+ 	"errors"
+ 	"fmt"
+ 	"hash"
++	"internal/goexperiment"
```

Comment:
I assume this is a MSFT only internal package where we have experimental functions/features etc?  Do we feature flag experimental features?

Reply:
`internal/goexperiment` is an internal package also available upstream, which basically contains feature flags for experimental features. We patch that package to add MSFT-only features. You can see the FIPS-related ones documented in here: https://github.com/microsoft/go/tree/microsoft/main/eng/doc/fips#usage-build